src/Controls/tests/DeviceTests/ControlsHandlerTestBase.iOS.cs

```diff
@@ -123,48 +119,55 @@ protected object GetTitleView(IElementHandler handler)
 
 		UIViewController[] GetActiveChildViewControllers(IElementHandler handler)
 		{
-			if (handler is IWindowHandler wh)
+			if (handler.VirtualView is IWindow window)
 			{
-				handler = wh.VirtualView.Content.Handler;
+				handler = window.Content.Handler;
 			}
 
-			if (handler is ShellRenderer renderer)
-			{
-				if (renderer.ChildViewControllers[0] is ShellItemRenderer sir)
-				{
-					return sir.SelectedViewController.ChildViewControllers;
-				}
-			}
+			if (handler.VirtualView is Page page)
+				handler = page.GetCurrentPage()?.Handler;
 
-			var containerVC = (handler as IPlatformViewHandler).ViewController;
-			var view = handler.VirtualView.Parent;
+			if (handler is null)
+				return new UIViewController[0];
 
-			while (containerVC is null && view is not null)
-			{
-				containerVC = (view?.Handler as IPlatformViewHandler).ViewController;
-				view = view?.Parent;
-			}
+			var navControllerResponder = (handler.PlatformView as UIView).FindResponder<UINavigationController>();
 
-			if (containerVC is null)
-				return new UIViewController[0];
+			if (navControllerResponder?.ChildViewControllers is not null)
+				return navControllerResponder.ChildViewControllers;
+
+			var contentResponder = (handler.PlatformView as UIView).FindResponder<ContainerViewController>();
 
-			return new[] { containerVC };
+			if (contentResponder?.NavigationController is not null)
+				return contentResponder.NavigationController.ChildViewControllers;
+
+			return new UIViewController[0];
 		}
 
 		UIViewController GetVisibleViewController(IElementHandler handler)
 		{
 			var vcs = GetActiveChildViewControllers(handler);
+			if (vcs.Length == 0)
+				return null;
```

Comment:
Should we change the return to `UIViewController?`

Reply:
we don't have nullable annotations enabled for this project